### PR TITLE
feat: added customer and supplier information to Batch

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
Ref: [Asana](https://app.asana.com/0/1192407896547052/1200001665851798)

### CODE REVIEW

**Review by neilLasrado — CHANGES_REQUESTED** [review]
I have shared feedback with @rvpasari

File: erpnext/stock/doctype/batch/batch.json [hunk 1]
```diff
@@ -134,9 +135,8 @@
   },
   {
    "fieldname": "supplier",
-   "fieldtype": "Link",
+   "fieldtype": "Data",
```
Comment (neilLasrado) [inline erpnext/stock/doctype/batch/batch.json]: @MyuddinKhatri did you change Supplier to a data field ?

**Review by neilLasrado — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/batch/batch.json (hunk 1 above)
Comment (MyuddinKhatri) [reply erpnext/stock/doctype/batch/batch.json]: Yes

File: erpnext/stock/doctype/stock_entry/stock_entry.py [hunk 2]
```diff
@@ -100,6 +100,7 @@ def on_submit(self):
 			self.update_so_in_serial_number()
 		self.update_package_tag()
 		self.update_package_tag_is_used()
+		self.update_batch_with_customer_providing_item()
```
Comment (neilLasrado) [inline erpnext/stock/doctype/stock_entry/stock_entry.py]: update_batch_with_customer_provided_item

File: erpnext/stock/doctype/stock_entry/stock_entry.py [hunk 3]
```diff
@@ -689,6 +690,16 @@ def update_stock_ledger(self):
 
 		self.make_sl_entries(sl_entries, self.amended_from and 'Yes' or 'No')
 
+	def update_batch_with_customer_providing_item(self):
```
Comment (neilLasrado) [inline erpnext/stock/doctype/stock_entry/stock_entry.py]: update_batch_with_customer_provided_item

**Review by neilLasrado — CHANGES_REQUESTED** [review]

**Review by neilLasrado — APPROVED** [review]

**Pull request merged**